modules/services/window-managers/fluxbox.nix

```
@@ -0,0 +1,135 @@
+        type = types.lines;
+        default = "";
+        description = ''
+          Keyboard shortcuts configuration for Fluxbox, written to
+          <filename>~/.fluxbox/keys<filename>. Look at the fluxbox-keys(1)
+          manpage for details.
+        '';
+      };
+
+      menu = mkOption {
+        type = types.lines;
+        default = "";
+        description = ''
+          Menu configuration for Fluxbox, written to
+          <filename>~/.fluxbox/menu<filename>. Look at the fluxbox-menu(1)
+          manpage for details.
+        '';
+      };
+
+      slitlist = mkOption {
+        type = types.lines;
+        default = "";
+        description = ''
+          Slitlist configuration for Fluxbox, written to
+          <filename>~/.fluxbox/slitlist<filename>. Look at the fluxbox(1)
+          manpage for details.
+        '';
+      };
+
+      windowmenu = mkOption {
+        type = types.lines;
+        default = "";
+        description = ''
+          Window menu configuration for Fluxbox, written to
+          <filename>~/.fluxbox/windowmenu<filename>. Look at the fluxbox-menu(1)
+          manpage for details.
+        '';
+      };
+
+      extraCommandLineArgs = mkOption {
+        type = with types; listOf str;
+        default = [ ];
+        example = [ "-log" "/tmp/fluxbox.log" ];
+        description = ''
+          Extra command line arguments to pass to <command>fluxbox</command>.
+          Look at the fluxbox(1) manpage for details.
+        '';
+      };
+
+      extraStartupCommands = mkOption {
+        type = with types; listOf str;
+        default = "";
+        example = "";
+        description = ''
+          Extra startup commands executed before <command>fluxbox</command>
+          starts.
+
+          Caution! Commands that should keep running after the window manager
+          starts should be appended with <literal>&</literal> to put them in the
+          background.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      (hm.assertions.assertPlatform "xsession.windowManager.fluxbox" pkgs
+        platforms.linux)
+    ];
+
+    home.packages = [ cfg.package ];
+
+    home.file = {
+      ".fluxbox/init" = mkIf (cfg.init != "") { text = cfg.init; };
+      ".fluxbox/apps" = mkIf (cfg.apps != "") { text = cfg.apps; };
+      ".fluxbox/keys" = mkIf (cfg.keys != "") { text = cfg.keys; };
+      ".fluxbox/menu" = mkIf (cfg.menu != "") { text = cfg.menu; };
+      ".fluxbox/slitlist" = mkIf (cfg.slitlist != "") { text = cfg.slitlist; };
+      ".fluxbox/windowmenu" =
+        mkIf (cfg.windowMenu != "") { text = cfg.windowmenu; };
+    };
+
+    xsession.windowManager.command = let
+      commandLine = concatStringsSep " " (remove ""
+        ([ "${fluxbox}/bin/fluxbox" "-rc ~/.fluxbox/init" ]
+          ++ (escapeShellArgs cfg.extraCommandLineArgs)));
+    in ''
+      ${cfg.extraStartupCommands}
+
+      exec ${commandLine} & fbpid=$!
+      wait $fbpid
```

Hmm, isn't this the same as

```suggestion
      ${commandLine}
```

?

---

Yes it is. I was following the advice from Fluxbox manual, that suggests this in order to accomodate post-start commands. 

But I will simplify this,

---

No, the exec will cause the rest of the script to be ignored since it performs a process substitution. See my other comment for a proposal on how to fix.